ams/ams-server/src/main/java/com/netease/arctic/ams/server/optimize/IcebergFullOptimizePlan.java

```diff
@@ -83,33 +84,30 @@ protected boolean partitionNeedPlan(String partitionToPath) {
     }
 
     Set<DeleteFile> partitionDeleteFiles = new HashSet<>();
-    Set<DataFile> partitionDataFiles = new HashSet<>();
     for (FileScanTask partitionFileScanTask : partitionFileScanTasks) {
-      partitionDataFiles.add(partitionFileScanTask.file());
       partitionDeleteFiles.addAll(partitionFileScanTask.deletes());
     }
 
     double deleteFilesTotalSize = partitionDeleteFiles.stream().mapToLong(DeleteFile::fileSizeInBytes).sum();
-    double dataFilesTotalSize = partitionDataFiles.stream().mapToLong(DataFile::fileSizeInBytes).sum();
-
-    long duplicateSize = PropertyUtil.propertyAsLong(arcticTable.properties(),
-        TableProperties.FULL_OPTIMIZE_TRIGGER_DUPLICATE_SIZE_BYTES_THRESHOLD,
-        TableProperties.FULL_OPTIMIZE_TRIGGER_DUPLICATE_SIZE_BYTES_THRESHOLD_DEFAULT);
-    double duplicateRatio = PropertyUtil.propertyAsDouble(arcticTable.properties(),
-        TableProperties.FULL_OPTIMIZE_TRIGGER_DUPLICATE_RATIO_THRESHOLD,
-        TableProperties.FULL_OPTIMIZE_TRIGGER_DUPLICATE_RATIO_THRESHOLD_DEFAULT);
-    // delete files total size reach target or delete files rate reach target
-    if (deleteFilesTotalSize > duplicateSize || deleteFilesTotalSize / dataFilesTotalSize >= duplicateRatio) {
+
+    long targetSize = PropertyUtil.propertyAsLong(arcticTable.properties(),
```

Comment:
Why do the `PropertyUtil` and `CompatiblePropertyUtil` work together? What's the difference?

Reply:
I used them together with carelessness, although It's safe to use `CompatiblePropertyUtil` here. I'll fix it.

If a new property is not compatible with any legacy property, there is no need to use `CompatiblePropertyUtil`.